crates/bitwarden/src/tool/generators/passphrase.rs

```diff
@@ -0,0 +1,240 @@
+    /// This value must be between 3 and 20.
+    pub num_words: u8,
+    /// Character separator between words in the generated passphrase. The value cannot be empty.
+    pub word_separator: String,
+    /// When set to true, capitalize the first letter of each word in the generated passphrase.
+    pub capitalize: bool,
+    /// When set to true, include a number at the end of one of the words in the generated passphrase.
+    pub include_number: bool,
+}
+
+impl Default for PassphraseGeneratorRequest {
+    fn default() -> Self {
+        Self {
+            num_words: 3,
+            word_separator: ' '.to_string(),
+            capitalize: false,
+            include_number: false,
+        }
+    }
+}
+
+const MINIMUM_PASSPHRASE_NUM_WORDS: u8 = 3;
+const MAXIMUM_PASSPHRASE_NUM_WORDS: u8 = 20;
+
+// We don't want the validated struct to be accessible, yet at the same time it needs to be public
+// to be used as a return type, so we define it in a private module to make it innaccessible.
+mod private {
+    /// Represents a set of valid options to generate a passhprase with.
+    /// To get an instance of it, use [`PassphraseGeneratorRequest::validate_options`](PassphraseGeneratorRequest::validate_options)
+    pub struct ValidPassphraseGeneratorOptions {
+        pub(super) num_words: u8,
+        pub(super) word_separator: String,
+        pub(super) capitalize: bool,
+        pub(super) include_number: bool,
+    }
+}
+use private::ValidPassphraseGeneratorOptions;
+
+impl PassphraseGeneratorRequest {
+    /// Validates the request and returns an immutable struct with valid options to use with the passphrase generator.
+    pub fn validate_options(self) -> Result<ValidPassphraseGeneratorOptions> {
+        // TODO: Add password generator policy checks
+
+        if !(MINIMUM_PASSPHRASE_NUM_WORDS..=MAXIMUM_PASSPHRASE_NUM_WORDS).contains(&self.num_words)
+        {
+            return Err(Error::Internal("'num_words' must be between 3 and 20"));
+        }
+
+        if self.word_separator.chars().next().is_none() {
+            return Err(Error::Internal("'word_separator' cannot be empty"));
+        };
+
+        Ok(ValidPassphraseGeneratorOptions {
+            num_words: self.num_words,
+            word_separator: self.word_separator,
+            capitalize: self.capitalize,
+            include_number: self.include_number,
+        })
+    }
+}
+
+/// Implementation of the random passphrase generator. This is not accessible to the public API.
+/// See [`ClientGenerator::passphrase`](crate::ClientGenerator::passphrase) for the API function.
+///
+/// # Arguments:
+/// * `options`: Valid parameters used to generate the passphrase. To create it, use
+///     [`PassphraseGeneratorRequest::validate_options`](PassphraseGeneratorRequest::validate_options).
+pub(super) fn passphrase(options: ValidPassphraseGeneratorOptions) -> String {
+    passphrase_with_rng(rand::thread_rng(), options)
+}
```

Comment:
I think the input here should be the `PassphraseGeneratorRequest`. The `ValidPassphraseGeneratorOptions` is a internal struct within the passphrase module so it doesn't make much sense having it as part of the public interface. This also removes the need to have a "private" module.